lib/Conversion/LowerDIP/LowerDIPPass.cpp

```diff
@@ -72,7 +73,17 @@ class DIPCorr2DOpLowering : public OpRewritePattern<dip::Corr2DOp> {
     auto boundaryOptionAttr = op.boundary_option();
     Value strideVal = rewriter.create<ConstantIndexOp>(loc, stride);
 
-    FloatType f32 = FloatType::getF32(ctx);
+    auto inElemTy = input.getType().cast<MemRefType>().getElementType();
+    auto kElemTy = kernel.getType().cast<MemRefType>().getElementType();
+    auto outElemTy = output.getType().cast<MemRefType>().getElementType();
+    auto constElemTy = constantValue.getType();
+    if (inElemTy != kElemTy || kElemTy != outElemTy || outElemTy != constElemTy) {
+        return op->emitOpError() <<  "input, kernel, output and constant must have the same element type";
```

Comment:
That makes sense. I will explore this.
However, likely, it would introduce some type conversion ops and bound checking, so it is better to make as a separate PR.

Reply:
So, since it requires adding a parameter (similar to opencv). I'd suggest making a separate issue to support it. I can create one. Is it ok?

Reply:
Okay

Reply:
@ArtemSkrebkov , can you please create a feature request so that we can keep track of this potential enhancement.

Reply:
Sure, done
https://github.com/buddy-compiler/buddy-mlir/issues/66

Seems, I am not able to set labels, so feel free to add some.